trymito.io/components/Glossary/PageTOC/PageTOC.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { useEffect, useState } from 'react';
+import { useDebouncedEffect } from '../../../hooks/useDeboundedEffect';
+import { classNames } from '../../../utils/classNames';
+import pageTOCStyles from './PageTOC.module.css';
+
+type Heading = {
+    level: string;
+    id: string;
+}
+
+const PageTOC = () => {
+    const [headings, setHeadings] = useState<Heading[]>([]);
+    const [activeHeading, setActiveHeading] = useState<Heading | undefined>(undefined);
+
+    useEffect(() => {
+        // Query all heading elements (h2, h3) in the content
+        const headingElements = document.querySelectorAll('h2, h3');
+    
+        // Extract text and level (e.g., "h1", "h2") from heading elements
+        const headingsArray: Heading[] = Array.from(headingElements).map((heading) => ({
+            level: heading.tagName.toLowerCase(),
+            id: heading.id,
+        })).filter((heading) => heading.id !== "");
+    
+        setHeadings(headingsArray);
+    }, []); // This useEffect only runs once on component mount
+    
+
+    useDebouncedEffect(() => {
```

Comment:
A debounced effect doesn't do what we want here. A debounced effect denounces running the effect when the [header] array changes. We don't want that -- we want to debounce the handleScroll function running when the scroll changes. 

To do so:
1. Remove the useDebouncedEffect (from trymito.io)
2. Debounce the handleScroll function from 'scroll' events

Google probably will be helpful here!